checker/src/body_visitor.rs

```diff
@@ -2169,4 +2199,39 @@ impl<'analysis, 'compilation, 'tcx, E> BodyVisitor<'analysis, 'compilation, 'tcx
             }
         }
     }
+
+    /// Extract the path and the value of the tag model field of the value located at `qualifier`.
+    /// If the tag model field is not tracked in the current environment, then either return an
+    /// unknown value (if `qualifier` is rooted at a parameter), or return a constant TRUE (if
+    /// `qualifier` is rooted at a local variable).
```

Comment:
The problem here is that we can only use `Variable { .. }` to represent a non-scalar value so we lose the information that a locally introduced struct does not have any tags. The value of a tag field is just a placeholder, so it seems fine to use any scalar type (bool, i32, etc.).

Reply:
It's similar to the default-model-field issue: because we cannot easily assign a non-scalar value, we want something that can indicate "there are no tags" as the default tag-field value for locally introduced structs.

Reply:
As you note yourself, the value of the tag field is just a placeholder. I don't quite see why it would be a problem to put a Variable value there instead of a bool. It is not a big deal, but it just seems less surprising to me.

Reply:
The problem is that we know nothing about the tags of a Variable value.

Reply:
Hmm. I am still having trouble with this. It actually now seems to me that the biggest problem here is that TRUE might have actual tags on it. Unlikely, for sure, but we have no defense. So, if we can perhaps make up a dedicated dummy untagged value that we are quite sure cannot be tagged (because it is not visible to the programmer), the code can be both more readable and safer too.